inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_primitive_fusing.cpp

```diff
@@ -811,9 +863,81 @@ void prepare_primitive_fusing::fuse_simple_primitives(program_impl &p) {
+                    node_queue.pop_front();
+                    if (std::find(node_history.begin(), node_history.end(), current_node.first) == node_history.end()) {
+                        node_history.push_back(current_node.first);
+                    }
+
+                    if (current_node.second > max_levels) {
+                        return;
+                    }
+
+                    // Push node to node_queue
+                    // If the node is already existed in node_queue, do not add it to the node_queue.
+                    auto push_node_queue = [&](cldnn::program_node* in_node, size_t level) {
+                        auto iter = std::find_if(node_queue.begin(), node_queue.end(), [&](std::pair<cldnn::program_node*, size_t> element) {
+                            return (in_node->id() == element.first->id());
+                        });
+                        if (iter == node_queue.end()) {
+                            node_queue.push_back(std::make_pair(in_node, level));
+                        }
+                    };
+
+                    // If the some user node is not eltwise and activation,
+                    // the current node will be considered as last node and put it back into the node_queue
+                    auto curr_users = current_node.first->get_users();
+                    auto invliad_user_iter = std::find_if(curr_users.begin(), curr_users.end(), [&](cldnn::program_node* user){
+                        auto ret = (user->is_output() ||
+                                (!(user->is_type<eltwise>() && user->get_primitive()->input.size() == 2) &&
+                                !(user->is_type<activation>() && user->get_primitive()->input.size() == 1)));
+                        return ret;
+                    });
+                    if (invliad_user_iter != curr_users.end()) {
+                        // If fused_node(i.e. Conv1) have invalid user node(that is not activation and eltwise ndoe), it cannot be fused
+                        if (fused_node->id() == current_node.first->id()) {
+                            return;
+                        }
+                        push_node_queue(current_node.first, (current_node.second+1));
+                        continue;
+                    }
+
+                    // Add user node in current node to the queue
+                    // But, do not add the node that passed once, it is checked using node_history
+                    for (auto& user : curr_users) {
+                        auto iter = std::find(node_history.begin(), node_history.end(), user);
+                        if (iter == node_history.end())
+                            push_node_queue(user, current_node.second+1);
+                    }
+                } while (node_queue.size() > 1);
+            } else {
+                merge_allowed = fused_node->get_users().size() == 1;
+            }
 
+            // This fusing can be extended to support peer node in any layout
```

Comment:
what do you mean with "any layout"? do you mean graph pattern?

Reply:
No, It's the oldest comments before implementing extended eltwise fusing. I will remove it

Reply:
I upload update commit. thanks